OpenChange/MAPIStoreMailFolder.m

```diff
@@ -631,17 +628,22 @@ - (BOOL) synchroniseCache
           if (!lastModseq
               || ([lastModseq compare: modseq] == NSOrderedAscending))
             lastModseq = modseq;
+
+          if ([[result objectForKey: @"flags"] containsObject: @"deleted"])
+            [currentProperties setObject: changeNumber
+                                  forKey: @"SyncLastDeleteChangeNumber"];
```

Comment:
I think we should check whether `changeNumber` is the latest in the same way is being performed for `SyncLastModseq` field.